app/services/katello/ui_notifications/pulp/proxy_disk_space.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  module UINotifications
+    module Pulp
+      class ProxyDiskSpace
+        class << self
+          def deliver!
+            SmartProxy.unscoped.with_features('Pulp').each do |proxy|
+              percentage = proxy.statuses[:pulp].storage['pulp_dir']['percent']
+              next unless percentage[0..2].to_i > 90
+              next if notification_already_exists?(proxy)
```

Comment:
i think that if its still exists, you should update the the expiry timestamps (e.g. if it happens over the weekend and no one looked, the 24 hours will pass before someone logs into satellite).